src/target/riscv/riscv.c

```diff
@@ -4766,7 +4766,7 @@ int riscv_init_registers(struct target *target)
 				case CSR_VL:
 				case CSR_VTYPE:
 				case CSR_VLENB:
-					r->exist = riscv_supports_extension(target, 'V');
+					r->exist = info->vlenb;
```

Comment:
Detail: To make the code easier to read, I'd recommend:

```suggestion
					r->exist = (info->vlenb > 0);
```

(There's one more occurrence below.)

Reply:
Changed. Thanks.